### ISSUE TITLE
Unable to create Lambda function using Terraform using existing IAM role. Error message is also not clear to understand.

### ISSUE THREAD

**Issue by sravan-bade** [hashicorp/terraform-provider-aws#21046]
Error receiving while creating Lambda function:
module.common.aws_lambda_function.log_monitoring_lambda_function: Creating...

Error: error creating Lambda Function (1): AccessDeniedException: 
        status code: 403, request id: d7aac889-0404-4234-b885-5608bbea4a97

Terraform script used:

Lambda.py

```terraform
data "aws_iam_role" "lambda_role" {
  name = "lambda-execution-role"
}

data "archive_file" "lambda_zip" {
  type        = "zip"
  source_file = "${path.module}/lambda_function.py"
  output_path = "${path.module}/lambda_log_monitoring_payload.zip"
}

resource "aws_lambda_function" "log_monitoring_lambda_function" {
  filename         = data.archive_file.lambda_zip.output_path
  source_code_hash = data.archive_file.lambda_zip.output_base64sha256
  function_name    = "${var.resource_name_prefix}-log-monitoring"
  role             = data.aws_iam_role.lambda_role.arn
  handler          = "lambda_function.lambda_handler"
  runtime          = "python3.8"

  environment {
    variables = {
      snsARN = aws_sns_topic.monitoring_topic.arn
    }
  }

}
```

**Comment by sravan-bade** [comment]
I tested manually with the lambda role and didn't face any issue there.

If this is related to created creation role issue then error message usually says, CreateLambdaFunction access is missing kind of message.

**Comment by justinretzolk** [comment]
Hey @sravan-bade 👋 Thank you for taking the time to file this issue. So that we have all of the information necessary to investigate this, can you update the issue with the rest of the information from the [issue template](https://github.com/hashicorp/terraform-provider-aws/blob/main/.github/ISSUE_TEMPLATE/Bug_Report.md)?

**Comment by willikisser** [comment]
Hi @sravan-bade I run exactly into the same problem. Do you have solved the problem in the meantime?

**Comment by willikisser** [comment]
> Hi @sravan-bade I run exactly into the same problem. Do you have solved the problem in the meantime?

I just solved our problem. 403 was really a 403. Fortunately, we activated AWS CloudTrail, and there we got the exact problem. We defined in our role policy which lambdas are allowed to create and we forgot to add it in the role policy. In case if the response returns more details about the missing permission, it would be nice if the error message is extended by the message. Thx.
